packages/firestore/src/core/sync_engine_impl.ts

```diff
@@ -1081,10 +1081,13 @@ async function applyDocChanges(
 
   const targetChange =
     remoteEvent && remoteEvent.targetChanges.get(queryView.targetId);
+  const waitForRequeryResult =
```

Comment:
nit: Could you rename `waitForRequeryResult` (here, and elsewhere in this PR)? I think the term "wait" throws me off because, to me, it suggests that the thread will block waiting for something to happen. Maybe a name like "targetIsPendingReset" or "targetHasExistenceFilterMismatches" would work? I think "targetIsPendingReset" is pretty good, actually. What do you think?